lazy_dataset/core.py

```diff
@@ -26,6 +26,29 @@ def _get_serialize_and_deserialize(immutable_warranty):
         raise ValueError(immutable_warranty)
 
 
+class NumpySerializedList(collections.UserList):
+    """
+    Stores the list dataset in two numpy arrays and behaves similar to a list
```

Comment:
Could you add a link to the block? We should give credits to the original author.